src/utils/shuffleArray.js

```diff
@@ -1,8 +1,10 @@
+// @ts-check
 //Fisher–Yates shuffle modern algorithm
 export default function shuffle(array) {
-  for (let i = array.length - 1; i > 0; i--) {
+  const arrayCopy = [...array]
```

Comment:
```suggestion
  array = [...array]
```
or
```suggestion
  array = array.slice()
```

Reply:
Be careful reassigning parameters. 
You can find good articles by searching for `js reassign parameter`.

The main issue is that you manipulate/mutate `arguments` here.